parts/k8s/cloud-init/artifacts/cse_config.sh

```diff
@@ -304,9 +304,10 @@ enableCRISystemdMonitor() {
 }
 {{- if NeedsContainerd}}
 installContainerd() {
+  removeMoby
   local v
   v=$(containerd -version | cut -d " " -f 3 | sed 's|v||')
-  if [[ $v != "${CONTAINERD_VERSION}" ]]; then
+  if [[ $v != "${CONTAINERD_VERSION}"* ]]; then
```

Comment:
we need this glob expansion because the versions we get from azure now include a `-azure` suffix